optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/io/XStreamConfigReader.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.solver.io;
+
+import com.thoughtworks.xstream.XStream;
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.config.solver.SolverConfig;
+
+public class XStreamConfigReader {
```

Comment:
New impl class, both solver and benchmark config reading use this class.

Reply:
I find this name good enough. But since it's a utility class with a private constructor, perhaps it should be `final`?

Reply:
done locally